1_wqp_pull/src/wqp_inventory.R

```diff
@@ -71,86 +72,101 @@ inventory_wqp <- function(state_id, wqp_pull_params) {
+  #
+  #   # first half of counties
+  #   wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[1:split]
+  #   temp1 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
+  #
+  #   # second half of counties
+  #   wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[(split+1):nrow(state_counties)]
+  #   temp2 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
+  #
+  #   # combine outputs
+  #   temp_dat <- list(time = temp1$time + temp2$time,
+  #                    nrow = nrow(temp1$out) + nrow(temp2$out),
+  #                    out = bind_rows(temp1$out, temp2$out))
+  # }
+
   # summarize what was retrieved and how long it took
   message('Retrieved ', temp_dat$nrow, ' rows of data in ', ceiling(temp_dat$time), ' seconds.')
-  
+
   return(temp_dat$out)
 }
 combine_inventory <- function(ind_file, ...) {
   dat_out <- bind_rows(...)
-  
+
   # keep columns of interest, put back lat/long
-  # keeping lat/long here in case there was a situation where you'd want to 
+  # keeping lat/long here in case there was a situation where you'd want to
   # take stock/map sites before pulling data
   dat_out <- dat_out %>%
-    select(OrganizationIdentifier, MonitoringLocationIdentifier, ResolvedMonitoringLocationTypeName, 
+    select(OrganizationIdentifier, MonitoringLocationIdentifier, ResolvedMonitoringLocationTypeName,
            StateName, CountyName, HUCEightDigitCode, latitude = lat, longitude = lon, resultCount)
 
   # write and indicate the data file
   data_file <- scipiper::as_data_file(ind_file)
   feather::write_feather(dat_out, data_file)
   gd_put(ind_file)
 }
-get_states <- function(states_url) {
-  
-  # get state and county codes
-  county_states <- dataRetrieval::countyCd %>%
-    mutate(state_id = paste0('US:', STATE))
-  
-  state_codes <- xml2::read_xml(states_url)
-  states <- unlist(as.character(xml2::xml_children(state_codes)))
-  states <- stringr::str_extract(states, pattern = 'US:\\d{2}')
-  states <- states[!is.na(states)]
-  
-  return(filter(county_states, state_id %in% states))
-}
+# get_states <- function(states_url) {
+#
+#   # get state and county codes
+#   county_states <- dataRetrieval::countyCd %>%
+#     mutate(state_id = paste0('US:', STATE))
+#
+#   state_codes <- xml2::read_xml(states_url)
+#   states <- unlist(as.character(xml2::xml_children(state_codes)))
+#   states <- stringr::str_extract(states, pattern = 'US:\\d{2}')
+#   states <- states[!is.na(states)]
+#
+#   return(filter(county_states, state_id %in% states))
+# }
 
 summarize_wqp_inventory <- function(inv_ind, out_file) {
- 
+
    wqp_inventory <- feather::read_feather(sc_retrieve(inv_ind, remake_file = 'getters.yml'))
-   
-   all <- data.frame(n_sites = nrow(wqp_inventory), 
+
+   all <- data.frame(n_sites = nrow(wqp_inventory),
                      n_records = sum(wqp_inventory$resultCount), stringsAsFactors = FALSE)
```

Comment:
Note that making calls by year is going to produce a different dataset than making calls by state. In the year call, sites will be represented in multiple rows. To make the summary (and more importantly the partitions for the data pulls), we need to collapse each site into a single row and take the sum of the resultCount.

Reply:
Thanks for that note, it is helpful.

Reply:
You'll also need to change the calculation for n_sites. This should now be the number of unique site IDs.